### PR TITLE
ci, Add ci using coreos ci

### PR DESCRIPTION
**Pull request by RamLavi**
Adding a new test script that will run related coreos ci tests
with the current set-ovs.sh script
Running this before each PR will help us achive better
resiliance and ci repo integrity

Signed-off-by: Ram Lavi <ralavi@redhat.com>

### CODE REVIEW

**Comment by RamLavi** [comment]
/hold
after https://github.com/coreos/coreos-assembler/pull/2221 is merged then we might want to add a test and also change the name of the current one.

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:109]:
```suggestion
fi
```

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:107]:
```suggestion
  echo "tests passed"
```

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:33]:
```suggestion
  if [[ -z "${latest_image_gz}" ]]; then
```

Let's keep the script consistent

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:41]:
```suggestion
  if [[ -z "${shasum}" ]]; then
```

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:47]:
```suggestion
  if [[ $(echo "${shasum} ${image_path}/${latest_image_gz}" | sha256sum --check) != "${image_path}/${latest_image_gz}: OK" ]]; then
```

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:85]:
```suggestion
    if [[ ! $(grep "PASS: ${test_name}" ${test_output}) ]]; then
```

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:103]:
```suggestion
if [[ ! -z ${tests_failed} ]]; then
```

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:5]: Please be more specific with the variable name

**Review by phoracek — CHANGES_REQUESTED** [review]
First iteration

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh:107]: Done

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh:33]: Done

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh:41]: Done

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh:47]: Done

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh:85]: Done

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh:103]: Done

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh]: Done

**Comment by RamLavi** [comment]
@phoracek first pass fixed [here](https://github.com/RHsyseng/rhcos-slb/compare/241b2b8a2903c1d7cd2739274375dae8f182ba5c..dc7925b5af919e77d36bea8a55547ab0c91d9285)

**Comment by RamLavi** [comment]
@phoracek is there a way to direct a project-presubmits job to a PR instead of the main branch, so that I could check the script lane before it is merged?

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh]: and Done :)

**Comment by phoracek** [comment]
> @phoracek is there a way to direct a project-presubmits job to a PR instead of the main branch, so that I could check the script lane before it is merged?

Well, a PR is just a branch on your fork, so you could setup Prow to test your fork ... but that sounds odd.

We usually setup CI for the project, so it is disabled by default and you have to trigger it with `/test`. That way it does not bother anybody.

File: tests/kola/test-coreos.sh
Comment (RamLavi) [inline tests/kola/test-coreos.sh:10]: I'm not 100% if that would be correct when running the prow job

**Comment by RamLavi** [comment]
updated tests on test list [here](https://github.com/RHsyseng/rhcos-slb/compare/f37c8081ee37ce306e240c100487522af50e2a9b..9b7cd0c698cfb888f307f1d6df28d459eb215bb1)

File: tests/kola/test-coreos.sh
Comment (phoracek) [inline tests/kola/test-coreos.sh:1]: Any reason to keep this under kola directory? Do you anticipate adding any other tests?

**Review by phoracek — CHANGES_REQUESTED** [review]
I tried to run this locally and it failed with following:

```
++ for test_name in "${TESTS_LIST[@]}"
++ test_output=/tmp/coreos-assembler-TvX0PF/rhcos.network.multiple-nics_output
++ ./bin/kola run -b rhcos --qemu-image /tmp/rhcos-latest-image/rhcos-4.7.13-x86_64-qemu.x86_64.qcow2 rhcos.network.multiple-nics
Error: harness: test suite failed
```

If we turn this into CI of this repo, it should provide better failure output to give us a chance to debug it.

File: tests/kola/test-coreos.sh
Comment (RamLavi) [reply tests/kola/test-coreos.sh:1]: I don't. I was keeping form with kola external test convention, but this is not really an external test in cosa terms, so we can totally loose this and just put it on tests folder.

**Comment by RamLavi** [comment]
> I tried to run this locally and it failed with following:
> 
> ```
> ++ for test_name in "${TESTS_LIST[@]}"
> ++ test_output=/tmp/coreos-assembler-TvX0PF/rhcos.network.multiple-nics_output
> ++ ./bin/kola run -b rhcos --qemu-image /tmp/rhcos-latest-image/rhcos-4.7.13-x86_64-qemu.x86_64.qcow2 rhcos.network.multiple-nics
> Error: harness: test suite failed
> ```
> 
> If we turn this into CI of this repo, it should provide better failure output to give us a chance to debug it.

I totally agree that we should expose this somehow.
kola tests offer a test folder with things like the test's journal.txt which is very helpful
If we use project-infra then I guess moving that folder to the artifacts folder should do it.
@kobi86 suggested running this ci on github Actions would be a better fit. It makes sense in a way, but then I need to explore how to store this output folder.

**Comment by kobi86** [comment]
I think we should: 
1. See if will be OK to run on Ubuntu or if we need fedora I can use my server) 
2. Create a CI for this using Github actions 
3. We can export the log file and expose it here in Github has artifact, just a step in the build:)

**Comment by RamLavi** [comment]
every test run takes ~800Kb, so currently every PR push will add 1.6Mb
although this is a rather quiet repo, I think we may reach the free limit (500Mb) pretty fast.

**Comment by RamLavi** [comment]
@phoracek added artifacts folder [here](https://github.com/RHsyseng/rhcos-slb/compare/9b7cd0c698cfb888f307f1d6df28d459eb215bb1..4dab594293ccb70ec6f1b0ceb64ef679967c8598)
I'll try to run it on a new fedora on beaker and see if I get anything

**Comment by RamLavi** [comment]
@phoracek fixed artifact copy command and added prerequisites so that the tests will pass on a clean fedora [here](https://github.com/RHsyseng/rhcos-slb/compare/4dab594293ccb70ec6f1b0ceb64ef679967c8598..4c315f182939a83de6302d90c3c592186d71e54a)
Can you retry to run it locally?

**Comment by RamLavi** [comment]
> @phoracek fixed artifact copy command and added prerequisites so that the tests will pass on a clean fedora [here](https://github.com/RHsyseng/rhcos-slb/compare/4dab594293ccb70ec6f1b0ceb64ef679967c8598..4c315f182939a83de6302d90c3c592186d71e54a)
> Can you retry to run it locally?

[update] - removed the prerequisites section to avoid running the script in sudo
Added a README section to assist local testing. 
@phoracek @kobi86 can you follow the README and tell me if it's clear&working on your local?

**Comment by RamLavi** [comment]
> > @phoracek fixed artifact copy command and added prerequisites so that the tests will pass on a clean fedora [here](https://github.com/RHsyseng/rhcos-slb/compare/4dab594293ccb70ec6f1b0ceb64ef679967c8598..4c315f182939a83de6302d90c3c592186d71e54a)
> > Can you retry to run it locally?
> 
> [update] - removed the prerequisites section to avoid running the script in sudo
> Added a README section to assist local testing [here](https://github.com/RHsyseng/rhcos-slb/compare/4c44f6a7184dcb8452937a94a623b29986b849c5..a785055ce5b9e55fdc40635f618b19824f661f11).
> @phoracek @kobi86 can you follow the README and tell me if it's clear&working on your local?

successfully tested on clean fedora
however I'm having trouble finding `qemu` and `swtpm` when trying it on a clean rhel8

**Review by phoracek — APPROVED** [review]
The test has passed. The README looks good. I would not bother adding RHEL 8 specific guidelines there.

**Review by kobi86 — CHANGES_REQUESTED** [review]

File: README.md
Comment (RamLavi) [reply README.md:56]: it actually doesn't matter where you run the script from right now

Comment (kobi86) [reply README.md:56]: Cool!

File: README.md
Comment (kobi86) [reply README.md:53]: @phoracek, I don't think we should add RHEL or other distributions to the readme, but I do think that we can add  an extra line that can make people life easy and instead wasting time to understand why it's not working they will know that this script with the command is meant to work of Fedora, if they would like to do it on other distributors then enjoy:)

Comment (RamLavi) [reply README.md:53]: ack, [Done](https://github.com/RHsyseng/rhcos-slb/compare/a785055ce5b9e55fdc40635f618b19824f661f11..789bcdd9804bac2d0bd56c55b6bac8c7dc38fd4b)

Comment (kobi86) [reply README.md:53]: Just to be sure, shouldn't it be Fedora?

Comment (RamLavi) [reply README.md:53]: totally - [Done](https://github.com/RHsyseng/rhcos-slb/compare/789bcdd9804bac2d0bd56c55b6bac8c7dc38fd4b..3858086bd73be3d684f65b84486f3b62931eadb4)

Comment (kobi86) [reply README.md:53]: Then LGTM:)

**Comment by RamLavi** [comment]
@kobi86 thanks for the lgtm, can you approve it in the PR, so that we could merge it?

**Pull request merged**
